Q/QuantumEspresso/bundled/patches/0000-pass-host-to-configure.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/install/extlibs_makefile b/install/extlibs_makefile
```

Comment:
It'd be good to upstream this patch.  Probably not as it is, because `MACHTYPE` and `target` are our own environment variables and `update_configure_scripts` is a [script we provide](https://docs.binarybuilder.org/stable/build_tips/#Utilities-in-the-build-environment), but the gist is that they have to
* pass correctly the configure flags around
* update their overly out-of-date autoconf helper scripts

Reply:
I agree ... I'll open an issue upstream pointing here.

Reply:
For reference, see https://gitlab.com/QEF/q-e/-/issues/341.

Reply:
It is quite unlikely to be accepted if an external script is needed especially when the script is intended for customizing stuff for a particular environment.

Reply:
Sorry, but I don't quite understand. Do you refer to the Issue I opened at the gitlab QE repository?

Reply:
@mfherbst yes. I saw `update_configure_scripts`

Reply:
That script simply updates the autoconf helper scripts.  QuantumEspresso developers should do that upstream.

Reply:
Ah yes, that's true, but this is only a workaround for cases where the autoconf helper scripts bundled upstream are not up to date (see [this documentation](https://docs.binarybuilder.org/stable/build_tips/#Utilities-in-the-build-environment) (and thus don't recognise certain architectures). So on the QE/FoX side one should be able to fix this by updating the autoconf build environment.

Reply:
https://gitlab.com/QEF/q-e/-/blob/e388497e9f6e95ec25b3c0e0b432dcceee86628d/install/config.guess#L29
https://gitlab.com/QEF/q-e/-/blob/e388497e9f6e95ec25b3c0e0b432dcceee86628d/install/config.sub#L35

Reply:
Actually, I think the ones that come with FoX also need to be updated as the `update_configure_scripts` runs inside the tree of the `fox` directory.

Reply:
OK. it is "not passed on" issue related to coupling multiple configure scripts and also some out-dated stuff gets in the way. QE autoconf maintainer might take a look. Is CMake an option in your build system?

Reply:
In principle yes. I tried it unsuccessfully, however and settled to autotools instead.

Reply:
Oh, yes [`update_configure_scripts`](https://github.com/JuliaPackaging/Yggdrasil/blob/c5ee8901595be390d7e56a60277f5172d00b7f1d/0_RootFS/Rootfs/bundled/utils/update_configure_scripts.sh) searches for all helper scripts recursively

Reply:
The CMake added 6.7 release was quite preliminary and doesn't work if you download a tar. But should be much better now in develop. The next release will be in the following weeks. So please report issues if the current develop has any for your needs.

Reply:
Sounds good. @giordano I could try the develop using a Git source (or what it's called) using our BinaryBuilder, right?

Reply:
[`GitSource`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.GitSource), yes

Reply:
Could you please explain

- where is `update_configure_scripts` and what it does?
- where are the _overly out-of-date autoconf scripts_?
- whether adding `--build=$(MACHTYPE) --host=$(target)` is guaranteed to always work?

Reply:
- The [`update_configure_scripts`](https://github.com/JuliaPackaging/Yggdrasil/blob/c5ee8901595be390d7e56a60277f5172d00b7f1d/0_RootFS/Rootfs/bundled/utils/update_configure_scripts.sh) is just a trick used by Yggdrasil to update the autoconf scripts. They are needed because the config.sub and config.guess scripts 
https://gitlab.com/QEF/q-e/-/blob/e388497e9f6e95ec25b3c0e0b432dcceee86628d/install/config.guess#L29
https://gitlab.com/QEF/q-e/-/blob/e388497e9f6e95ec25b3c0e0b432dcceee86628d/install/config.sub#L35
are quite old and don't recognise newer platforms. I have not checked, but I suspect that the similarly old scripts are located inside the FoX build tree and need to be updated as well.
- Adding `--build=$(MACHTYPE) --host=$(target)` has been my hack to solve this problem for us (where `MARCHTYPE` and `target` are Yggdrasil-specific environment variables). For the general case in a QE build one would need to get the values passed to `--build` and `--host` to QE's `configure` script and pass them on to FoX's `configure` in the `extlibs_makefile`. I honestly have no clue how one would do that.

Reply:
I see. The config.sub and config.guess files in QE are now updated. About FoXI noticed this: https://github.com/andreww/fox/pull/70. Passing additional configure arguments to the configure of external libraries is feasible, I guess, by writing them somewhere in the 'make.inc' file produced by configure, but this requires some though.